BitList.cs

```diff
@@ -0,0 +1,1496 @@
+        {
+            ToUInt16Array().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(uint[] array, int arrayIndex)
+        {
+            ToUInt32Array().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(ulong[] array, int arrayIndex)
+        {
+            ToUInt64Array().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(float[] array, int arrayIndex)
+        {
+            ToSingleArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(double[] array, int arrayIndex)
+        {
+            ToDoubleArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(decimal[] array, int arrayIndex)
+        {
+            ToDecimalArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Copy this instance to the specified array.
+        /// </summary>
+        /// <param name="arrayIndex">Start position.</param>
+        public void CopyTo(char[] array, int arrayIndex)
+        {
+            ToCharArray().CopyTo(array, arrayIndex);
+        }
+
+        /// <summary>
+        /// Get a the value in the specified index.
+        /// </summary>
+        /// <exception cref="IndexOutOfRangeException" />
+        public bool Get(int index)
+        {
+            if (!IsEmpty && index < bits.Length && index >= 0)
+                return bits[index];
+            else
+                throw new IndexOutOfRangeException();
+        }
+
+        /// <summary>
+        /// Get a range of value in the specified index.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException" />
+        public BitList Get(int index, int count)
+        {
+            if (!IsEmpty && index < bits.Length && index >= 0 && index + count <= bits.Length - index && count > 0)
+            {
+                var bools = new List<bool>();
+                for (int i = 0; i < count; i++)
+                    bools.Add(bits[index + i]);
+                return new BitList(bools.ToArray());
+            }
+            else
+                throw new ArgumentOutOfRangeException();
```

Comment:
Codacy found an issue: [Use a constructor overloads that allows a more meaningful exception message to be provided.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)